source/agora/consensus/protocol/Nominator.d

```diff
@@ -1257,7 +1271,8 @@ extern(D):
                     hash: data.hashFull(),
                     total_rate = total_rate,
                 };
-                candidate_holders ~= candidate_holder;
+                if (!skip_candidate)
```

Comment:
This still does not solve the root cause here, nodes still can select different candidates and that is probably perfectly normal.

Before they could select different ones because they calculate the fees different, now they can still do the same because they have different TXs in their pool. I think it was less likely to happen before, because a single TX that we don't include in fee calculations would have less of an impact.

We already have a TX catchup process anyways, missing TXs should not delay block creation indefinitely.

Reply:
<img width="1808" alt="Screen Shot 2021-11-01 at 16 22 39" src="https://user-images.githubusercontent.com/11815575/139636226-931c0a5d-c746-40d5-9920-d22f1c424a07.png">

Reply:
^ This is from the nodes in Production that are not slashed. This pattern has repeated for over a week now.

Reply:
It looks like v2 has a tx which the others are not able to get during catchup.